### DISCUSSION TITLE
Retransmission of Confirmed uplinks

### DISCUSSION THREAD

**Discussion by Adamzat** [Lora-net/LoRaMac-node#1096] [Q&A] [answered]
Hi all,

I have a question about the retransmission mechanism used for confirmed uplink frames.
 
With the old version of LoraWan Stack 4.4.0 (LoraWan Spec. 1.0.2), an uplink is retransmitted Maximum 8 times (MAX_ACK_RETRIES = 8 ) until an Ack is received from the Network Server. 

However, this automatic mechanism seems to be deleted in LoraWan Stack 4.5.1 (LoraWan 1.0.4) and a frame is sent only one. The retransmission can be managed by the application Layer.

Can you please confirm that is an expected behavior ? Is there a description about this mechanism in the LoraWan specification 1.0.4.

Thank you.

Best regards.

**Comment by mluis1** [comment, marked as answer]
LoRaWAN 1.0.4 specification changed the way Confirmed uplinks re-transmissions are handled.
In 1.0.4 the Confirmed uplinks are handled in the same way as Unconfirmed uplinks. 

In case the acknowledgement is not received it is up to the application layer to transmit a new Confirmed uplink.

Please refer to below [LoRaWAN 1.0.4](https://lora-alliance.org/resource-hub/lorawan-104-specification-package) specification chapters.
- 4.3.1.2 Frame acknowledge bit and acknowledgment procedure (ACK in FCtrl) 
- 4.3.1.3 Retransmission procedure
- 16.2 Uplink Timing Diagram for Confirmed Data Frames

**Reply by mluis1** [reply]
In previous LoRaWAN specifications you can almost mimic the same behavior by setting the number of trials to 1.

The difference will be the `LinkAdrReq` mac command  `NbTrans` parameter which was only applied to Unconfirmed uplink in previous LoRaWAN specifications. While in 1.0.4 it is applied to Unconfirmed and Confirmed uplinks.

**Reply by Adamzat** [reply]
Thank you Miguel,
It is very clear.

Best regards.

**Reply by mluis1** [reply]
You are welcome.
